src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+        return the actual value used
+        """
+        # Pick the resolution which matches the scale in X
+        res_x = int(round(self._shape[0] / value[0]))
+        res = next(r for r in self.resolution.choices if r[0] == res_x)
+
+        # TODO: instead of setting both X and Y, only set X, and read back Y?
+        # This would be slightly more flexible in case the XT lib supports other
+        # resolutions than the hard-coded ones. For now we assume the hard-coded
+        # ones are all the possibles ones.
+        self.parent.set_resolution(res, channel=self.channel)
+        self.resolution._set_value(res, force_write=True)
+
+        return value
+
+    def _onScale(self, s):
+        self._updatePixelSize()
+
+    def _updateResolution(self):
+        """
+        To be called to read the server resolution and update the corresponding VAs
+        """
+        resolution = tuple(self.parent.get_resolution(self.channel))
+        if resolution != self.resolution.value:
+            scale = (self._shape[0] / resolution[0],) * 2
+            self.scale._value = scale  # To not call the setter
+            self.resolution._set_value(resolution, force_write=True)
+            self.scale.notify(scale)
+
+    def _updatePixelSize(self):
+        """
+        Update the pixel size using the horizontalFoV.
+        """
+        fov = self.horizontalFoV.value
+        # The pixel size is equal in x and y.
+        pxs = (fov / self._shape[0],
+               fov / self._shape[0])
+        # pixelSize is read-only, so we change it only via _value
+        self.pixelSize._value = pxs
+        self.pixelSize.notify(pxs)
+        # If scaled up, the pixels are bigger
+        pxs_scaled = (pxs[0] * self.scale.value[0], pxs[1] * self.scale.value[1])
+        self._metadata[model.MD_PIXEL_SIZE] = pxs_scaled
+
+    def _setDwellTime(self, dwell_time):
+        self.parent.set_dwell_time(dwell_time, channel=self.channel)
+        return self.parent.get_dwell_time(self.channel)
+
+    def _setVoltage(self, voltage: float) -> float:
+        self.parent.set_high_voltage(voltage, channel=self.channel)
+        return self.parent.get_high_voltage(self.channel)   # return the actual value used
+
+    def _setCurrent(self, current: float) -> float:
+        self.parent.set_beam_current(current, channel=self.channel)
+        return self.parent.get_beam_current(self.channel)  # return the actual value used
+
+    def _setBlanker(self, blank: bool) -> bool:
+        """
+        Parameters
+        ----------
+        blank (bool or None): True if the the electron beam should blank, False if it should be unblanked,
+            None if it should be blanked/unblanked automatically.
+
+        Returns
+        -------
+        (bool or None): True if the the beam is blanked, False if it is unblanked. See Notes for edge case,
+            None if it should be blanked/unblanked automatically.
```

Comment:
The beam blanking is a bit annoying. If you have trouble with it during testing it might be good to have look at this https://drive.google.com/file/d/1f5xdaA16qMMYpCGvDacCrIpUKAhRsu3Y/view (I can also explain the diagram if necessary)